source/extensions/common/dynamic_forward_proxy/dns_cache_impl.h

```diff
@@ -14,9 +14,29 @@ namespace Extensions {
 namespace Common {
 namespace DynamicForwardProxy {
 
+/**
+ * All DNS cache stats. @see stats_macros.h
+ */
+#define ALL_DNS_CACHE_STATS(COUNTER, GAUGE)                                                        \
+  COUNTER(dns_query_attempt)                                                                       \
+  COUNTER(dns_query_success)                                                                       \
+  COUNTER(dns_query_failure)                                                                       \
+  COUNTER(host_address_changed)                                                                    \
+  COUNTER(host_added)                                                                              \
+  COUNTER(host_removed)                                                                            \
+  GAUGE(num_hosts, Accumulate)
```

Comment:
Is the dns cache passed through hot-restart? `NeverImport` for this? cc @jmarantz

Reply:
Accumulate seemed more natural to me for this, but I don't feel strongly about it.

Reply:
Correct me if I'm wrong, if a hot-restart happened when there is entries in dns cache, IIUC the new process will start with a new empty dns cache anyway, so if we transfer the gauge, then the stats won't reflect number of hosts in the cache anymore?

Reply:
The stat would be the sum of both caches, which is arguably what you want, but like I said I could go either way on this.

Reply:
I think if we were going to track dns-cache hits I'd want that to be Accumulate.

But if this is the number of hosts currently in the dns-cache I think we want NeverImport unless we transfer the cache contents.

Reply:
> But if this is the number of hosts currently in the dns-cache I think we want NeverImport unless we transfer the cache contents.

We do accumulate a bunch of things that we don't transfer, like connections and requests, so I don't think it's an awful idea to do that here since we are roughly modeling the total cache/memory size across both processes, but like I said I don't feel strongly about it so I will switch it to never import.